src/ProfileEndpoints.jl

```diff
@@ -315,6 +316,28 @@ function handle_task_backtraces(stage_path = nothing)
     return HTTP.Response(200, backtrace_file)
 end
 
+###
+### Profile removal
+###
+
+function has_profile_extension(file)
+    return endswith(file, ".profile") || endswith(file, ".pb.gz") || endswith(file, ".task_backtraces")
+end
+
+function handle_profile_removal(profile_file, stage_path = nothing)
+    if stage_path !== nothing
+        profile_file = joinpath(stage_path, profile_file)
+    end
+    if !has_profile_extension(profile_file)
+        return HTTP.Response(400, "Profile file must have a `.profile`, `.pb.gz`, or `.task_backtraces` extension")
+    end
+    if !isfile(profile_file)
+        return HTTP.Response(400, "Profile file not found")
+    end
+    rm(profile_file)
```

Comment:
I'm still worried that exposing a `rm` through an endpoint can be exploited somehow and I wonder how we can make this as safe as possible. Maybe let's check with someone from security or @NHDaly? 

I think ideally we'd keep a list of files that we created and that we can delete safely... or something else that makes this very predictable and safe.

Reply:
Resolved in last commit in which we keep track of files created in a Set.